praw/models/modaction.py

```diff
@@ -9,8 +11,19 @@ class ModAction(PRAWBase):
     @property
     def mod(self):
         """Return the Redditor who the action was issued by."""
-        return self._reddit.redditor(self._mod)
+        return self._data['mod']
 
-    @mod.setter
-    def mod(self, value):
-        self._mod = value  # pylint: disable=attribute-defined-outside-init
+    @classmethod
+    def _objectify_acknowledged(cls, reddit, data):
+        key = 'mod'
+        item = data.get(key)
+        if isinstance(item, string_types):
+            data[key] = Redditor(reddit, name=item)
+        elif isinstance(item, Redditor):
```

Comment:
Previously it seems we simply always passed the instance into `reddit.Redditor` and didn't check the type. I suppose this saves on creating a new instance of Redditor.

I'm also curious why `_reddit` needs to be set. It seems like that should be set at the moment that `Redditor` is created.

I also wonder if the `elif` should simply be an `else` as I don't know what other types this could take... maybe `none` if `mod` doesn't exist.

Reply:
Setting `_reddit` here isn’t actually needed for PRAW to work correctly. The PRAW codebase never actually makes use of this code path.

I just did this specifically to solve issue #954. It can’t hurt to do so, right?

> I also wonder if the `elif` should simply be an `else` as I don't know what other types this could take... maybe `none` if `mod` doesn't exist.

Yes, it could be `None` so that’s why it’s `elif`.